### DISCUSSION TITLE
Unable to add messaging().setBackgroundMessageHandler in index.js for Android release build

### DISCUSSION THREAD

**Discussion by pschill** [invertase/react-native-firebase#4165] [Q&A]
Hello!

I've got a really strange error that is literary driving me nuts.

I want to listen for background messages in my react-native application (expo bare workflow) and thus added messaging().setBackgroundMessageHandler in my index.js.

Everything works in debug mode but the listener is not triggered when I build the application with --variant=release.

I tried react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res which gives me no visible error. But it really seems like the index.js is not getting fired in the release mode for Android. iOS works like a charm.

I cleaned the project with gradle at least 100 times, expo publish 200 times and tried to upload the app to app store etc. etc. but nothing works.

The listener works if I put it in App.js, but ofcourse not when the app is killed and it is really important that the listener gets triggered all the time since I'm building a calling app.

index.js
```
import App from './App';
import React from "react";
import RNCallKeep from "react-native-callkeep";
import {setupAndroidNotifications, setupIosNotifications} from "./misc/Pushnotifications";
import {Platform} from "react-native";
import {registerRootComponent} from 'expo';


if (Platform.OS === 'android') {
    console.info('setup android');
    setupAndroidNotifications((data) => {
            console.log('initiate incoming call: ' + JSON.stringify(data));
            const credentials = JSON.parse(data.credentials);
            RNCallKeep.displayIncomingCall(data.uuid, data.handle, data.callerName, credentials.apiKey, credentials.sessionId, credentials.token);
        });
} else if (Platform.OS === 'ios') {
    setupIosNotifications((data) => {
        console.info('got token'+ data);
    });
}
registerRootComponent(App);
```
setupAndroidNotifications function
```
export const setupAndroidNotifications = async (notificationCallback) => {
    //firebase messaging
    messaging().onMessage(async remoteMessage => {
        console.log("message: " + remoteMessage);
        const credentials = remoteMessage.data;
        if (credentials) {
            notificationCallback(credentials);
        }
    });

    messaging().setBackgroundMessageHandler(async remoteMessage => {
        console.log("cred: " + remoteMessage)
        const credentials = JSON.parse(JSON.stringify(remoteMessage.data));
        if (credentials) {
            notificationCallback(credentials);
        }
    });
};
```

Thanks,
Philip

**Comment by Salakar** [comment]
> but ofcourse not when the app is killed and it is really important that the listener gets triggered all the time since I'm building a calling app.

Can you clarify what you mean by `app is killed` - what steps are you taking to kill the app, swipe away or force quit in app settings?
